contrib/format-msgpack/src/main/java/org/apache/drill/exec/store/msgpack/MsgpackFormatPlugin.java

```diff
@@ -0,0 +1,212 @@
+    return super.getMatcher();
+  }
+
+  @Override
+  public RecordReader getRecordReader(FragmentContext context, DrillFileSystem dfs, FileWork fileWork,
+      List<SchemaPath> columns, String userName) {
+    return new MsgpackRecordReader(getConfig(), context, fileWork.getPath(), dfs, columns);
+  }
+
+  @Override
+  public RecordWriter getRecordWriter(FragmentContext context, EasyWriter writer) throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @JsonTypeName("msgpack")
+  public static class MsgpackFormatConfig implements FormatPluginConfig {
+
+    public List<String> extensions = ImmutableList.of("mp");
+    private boolean readBinaryAsString = false;
+    private boolean lenient = true;
+    private boolean printToConsole = true;
+    private boolean learnSchema = true;
+    private boolean useSchema = true;
+    private static final List<String> DEFAULT_EXTS = ImmutableList.of("mp");
+
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    public List<String> getExtensions() {
+      if (extensions == null) {
+        // when loading an old JSONFormatConfig that doesn't contain an
+        // "extensions"
+        // attribute
+        return DEFAULT_EXTS;
+      }
+      return extensions;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isReadBinaryAsString() {
+      return readBinaryAsString;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isLenient() {
+      return lenient;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isPrintToConsole() {
+      return printToConsole;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isLearnSchema() {
+      return learnSchema;
+    }
+
+    @JsonInclude(JsonInclude.Include.ALWAYS)
+    public boolean isUseSchema() {
+      return useSchema;
+    }
+
+    public void setUseSchema(boolean useSchema) {
+      this.useSchema = useSchema;
+    }
+
+    public void setLearnSchema(boolean learnSchema) {
+      this.learnSchema = learnSchema;
+    }
+
+    public void setExtensions(List<String> extensions) {
+      this.extensions = extensions;
+    }
+
+    public void setLenient(boolean lenient) {
+      this.lenient = lenient;
+    }
+
+    public void setReadBinaryAsString(boolean readBinaryAsString) {
+      this.readBinaryAsString = readBinaryAsString;
+    }
+
+    public void setPrintToConsole(boolean printToConsole) {
+      this.printToConsole = printToConsole;
+    }
+
+    @Override
+    public int hashCode() {
+      final int prime = 31;
+      int result = 1;
+      result = prime * result + ((extensions == null) ? 0 : extensions.hashCode());
+      result = prime * result + (learnSchema ? 1231 : 1237);
+      result = prime * result + (printToConsole ? 1231 : 1237);
+      result = prime * result + (lenient ? 1231 : 1237);
+      result = prime * result + (readBinaryAsString ? 1231 : 1237);
+      result = prime * result + (useSchema ? 1231 : 1237);
```

Comment:
There is a handy Guava class for this. Example from this [sample plugin](https://github.com/paul-rogers/drill/blob/5e5dc8652bac7750be075154488394cadc32341d/contrib/format-regex/src/main/java/org/apache/drill/exec/store/easy/regex/RegexFormatConfig.java):

```
  @Override
  public boolean equals(Object obj) {
    if (this == obj) { return true; }
    if (obj == null || getClass() != obj.getClass()) {
      return false;
    }
    final RegexFormatConfig other = (RegexFormatConfig) obj;
    return Objects.equal(regex, other.regex) &&
           Objects.equal(fields, other.fields) &&
           Objects.equal(extension, other.extension);
  }
   @Override
  public int hashCode() {
    return Arrays.hashCode(new Object[] {regex, fields, extension});
  }
```

Now, I have no idea why we hash or compare these objects since doing so makes little sense...